backends/cadence/fusion_g3/operators/op_dequantize.cpp

```diff
@@ -0,0 +1,810 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include <executorch/kernels/portable/cpu/util/reduce_util.h>
+#include <executorch/runtime/kernel/kernel_includes.h>
+#include <xa_nnlib_kernels_api.h>
+#include <algorithm>
+#include <cinttypes>
+#include <cmath>
+
+using exec_aten::Scalar;
+using exec_aten::ScalarType;
+using exec_aten::Tensor;
+using torch::executor::Error;
+using torch::executor::KernelRuntimeContext;
+
+template <typename T>
+using optional = exec_aten::optional<T>;
+/* ScalarType in Executorch do not have support for below data types.
+ * So, creating a placeholder for these data types. Once, ScalarTypes is
+ * updated to have support for below data types, these can be removed and 
+ * operator need to be updated accordingly
+ */
+ 
+ enum datatype {
+ Ushort = 20,
+ Bits4u = 21,
+ Bits4  = 22
+ };
+
+/**
+ * For an input tensor, use the scale and zero_point arguments to quantize it.
+ */
+namespace cadence { 
+namespace impl {
+namespace G3 {
+namespace native {
+
+namespace {
+
+/**
+ * Asserts that the parameters are valid.
+ */
+void check_dequantize_per_tensor_args(const Tensor& input,
+                                    int64_t quant_min,
+                                    int64_t quant_max,
+                                    ScalarType dtype,
+                                    exec_aten::optional<ScalarType>& out_dtype,
+                                    Tensor& out)
+{
+    ET_CHECK_MSG(
+      input.scalar_type() == ScalarType::Byte ||
+          input.scalar_type() == ScalarType::Char ||
+          input.scalar_type() == ScalarType::Bits16 ||
+          input.scalar_type() == ScalarType::Short ||
+          input.scalar_type() == (ScalarType) Ushort ||
+          input.scalar_type() == (ScalarType) Bits4 ||
+          input.scalar_type() == (ScalarType) Bits4u ||
+          input.scalar_type() == ScalarType::Int,
+          
+      "input.scalar_type() %" PRId8 " is not supported:",
+      static_cast<int8_t>(input.scalar_type()));
+
+    ET_CHECK_MSG(
+      input.scalar_type() == dtype,
+      "input.scalar_type() %" PRId8 " is not matching dtype argumenta:",
+      static_cast<int8_t>(input.scalar_type()));
+
+    if (out_dtype.has_value()) {
+    ET_CHECK_MSG(
+        out.scalar_type() == out_dtype.value(),
+        "output_dtype must match the dtype of the out tensor");
+    }
+
+    ET_CHECK_MSG(
+      quant_min <= quant_max,
+      "quant min: %" PRId64 " is greater than quant max: %" PRId64,
+      quant_min,
+      quant_max);
+}
+
+} // namespace
+
+
+/* Local function which calls the kernels based on the input datatype */
+void Dequantize_impl(Tensor& out,
```

Comment:
please use all lower case  fixed in https://github.com/pytorch/executorch/pull/7061